common/frame_session.go

```diff
@@ -1084,3 +1084,18 @@ func (fs *FrameSession) updateViewport() error {
 
 	return nil
 }
+
+func (fs *FrameSession) executionContextForID(
+	executionContextID cdpruntime.ExecutionContextID,
+) (*ExecutionContext, error) {
+	fs.contextIDToContextMu.Lock()
+	defer fs.contextIDToContextMu.Unlock()
+
+	for id, exc := range fs.contextIDToContext {
+		if id == executionContextID {
+			return exc, nil
+		}
+	}
```

Comment:
Why are we ranging over the `contextIDToContext` map?

Reply:
Oh, I missed this 👍 I also wonder why :)

Reply:
Force pushed applying the change at [4136588](https://github.com/grafana/xk6-browser/pull/1009/commits/4136588f95c2462bb64caf9b05132b23f436af5f)